### PR TITLE
Document shell.env json serialization behaviour.

### PR DESCRIPTION
**Pull request by neykov**
Depends on https://github.com/apache/brooklyn-server/pull/216

### CODE REVIEW

File: guide/yaml/example_yaml/vanilla-bash-netcat-env.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+location: localhost
+services:
+- type: org.apache.brooklyn.entity.software.base.VanillaSoftwareProcess
+  name: Simple Netcat Server
+  launch.command: |
+    echo $MESSAGE | nc -l $NETCAT_PORT &
+    echo $! > $PID_FILE
+    
+  env:
+    MESSAGE: hello
```
Comment (geomacy) [inline guide/yaml/example_yaml/vanilla-bash-netcat-env.yaml]: maybe worth adding a $brooklyn.config(....) in as the value, to emphasise the reusability mentioned in the .md, here and in the port.

File: guide/yaml/custom-entities.md [hunk 2]
```diff
@@ -82,6 +82,18 @@ to be specified up one level, alongside the type.
 However config keys which are *not* declared on the type *must* be declared in the `brooklyn.config` block. 
 
 
+### Passing custom variables
+
+Blueprint scripts can be parametrized through environment variables, making them reusable in different use-cases.
```
Comment (geomacy) [inline guide/yaml/custom-entities.md]: typo "parametrised"

**Comment by geomacy** [comment]
Looks good; short but sweet.  A couple of minor comments.

File: guide/yaml/custom-entities.md [hunk 3]
```diff
@@ -82,6 +82,18 @@ to be specified up one level, alongside the type.
...
+Define the variables in the `env` block and then reference them using the standard bash notation:
```
Comment (geomacy) [inline guide/yaml/custom-entities.md]: or 'shell.env' (is there any difference? what's preferred?)

Comment (neykov) [inline guide/yaml/custom-entities.md]: Used env to be consistent with the other examples on the page. "env" is used when not using "brooklyn.config". "shell.env" is used when nested in "brooklyn.config" (that's the original idea, currently they are more or less interchangeable).

We should be consistently using "brooklyn.config" but that's for another PR.

File: guide/yaml/example_yaml/vanilla-bash-netcat-env.yaml (hunk 1 above)
Comment (neykov) [inline guide/yaml/example_yaml/vanilla-bash-netcat-env.yaml]: Parameters and DSL are introduced in the next section. Want to keep it short and simple and up to the point here.

**Comment by neykov** [comment]
Thanks @geomacy - fixed the typo and merging.

**Pull request merged**